File: IMPLEMENTATION/1_LRUD/lrud.cpp
#include <bits/stdc++.h>

using namespace std;

int n;
string plans;

int x = 1;
int y = 1;

int dx[4] = {0, 0, -1, 1};
int dy[4] = {-1, 1, 0, 0};
char moveTypes[4] = {'L', 'R', 'U', 'D'};

int main(void)
{
    cin >> n;
    cin.ignore(); // 정수 다음에 문자열이 주어질 때는 버퍼를 지워주어야 한다.
    getline(cin, plans);

    for (int i = 0; i < plans.size(); i++)
    {
        char plan = plans[i];
        int nx = 0;
        int ny = 0;
        for (int j = 0; j < 4; j++)
        {
            if (moveTypes[j] == plan)
            {
                nx = x + dx[j];
                ny = y + dy[j];
                break;
            }
        }
        if (nx < 1 || ny < 1 || nx > n || ny > n)
            continue;
        x = nx;
        y = ny;
    }
    cout << x << ' ' << y << endl;
    return 0;
}

File: IMPLEMENTATION/1_LRUD/lrud.py
n = int(input())
plans = list(map(str, input().split()))

x, y = 1, 1
dx = [0, 0, -1, 1]
dy = [-1, 1, 0, 0]
move_types = ['L', 'R', 'U', 'D']

for plan in plans:
    nx, ny = 0, 0
    for i in range(len(move_types)):
        if plan == move_types[i]:
            nx = x + dx[i]
            ny = y + dy[i]
            break
    if nx < 1 or ny < 1 or nx > n or ny > n:
        continue
    x = nx
    y = ny

print(x, y)

# x, y = 1, 1
# dx, dy = [], []

# for d in plans:
#     if d == "L":
#         dx.append(0)
#         dy.append(-1)
#     elif d == "R":
#         dx.append(0)
#         dy.append(1)
#     elif d == "U":
#         dx.append(-1)
#         dy.append(0)
#     elif d == "D":
#         dx.append(1)
#         dy.append(0)

# for i in range(len(plans)):
#     nx = x + dx[i]
#     ny = y + dy[i]
#     if nx >= 1 and nx <= n and ny >= 1 and ny <= n:
#         x = nx
#         y = ny

# print(x, y)